### PR TITLE
fix(#15241): removed invalid transforms on Inputs

### LINKED ISSUE #15241: Component: P-Table does work correctly with  [rowTrackBy] for 17.13.0 version
### Describe the bug

When use we `"primeng": "17.13.0" ` and `<p-table [rowTrackBy]="rowTrackBy" ....></p-table>`

```
  rowTrackBy(_index, item) {
    return item.id;
  }
```

We have warning for develop and tests env: 
`'trackBy must be a function, but received null. See https://angular.io/api/common/NgForOf#change-propagation for more information.'`

And it really doesn't use in `p-table`. 

But when we use `"primeng": "17.12.0" ` or less it works correctly

### Environment

```
{
  "name": "frontend",
  "private": true,
  "dependencies": {
    "@angular/animations": "17.3.3",
    "@angular/cdk": "17.3.3",
    "@angular/common": "17.3.3",
    "@angular/compiler": "17.3.3",
    "@angular/core": "17.3.3",
    "@angular/forms": "17.3.3",
    "@angular/platform-browser": "17.3.3",
    "@angular/platform-browser-dynamic": "17.3.3",
    "@angular/router": "17.3.3",
    "@ctrl/ngx-emoji-mart": "9.2.0",
    "@ng-select/ng-select": "12.0.7",
    "@ngx-translate/core": "15.0.0",
    "@rails/actioncable": "7.0.8",
    "@rails/activestorage": "7.0.8",
    "@types/jquery": "3.5.29",
    "@typescript-plus/fast-memoize-decorator": "0.2.1",
    "angular-mentions": "1.5.0",
    "autolinker": "4.0.0",
    "bootstrap": "4.6.2",
    "chart.js": "4.4.2",
    "chartjs-plugin-datalabels": "2.2.0",
    "dayjs": "1.11.10",
    "deepmerge-ts": "5.1.0",
    "easymde": "2.18.0",
    "font-awesome": "4.7.0",
    "jquery": "3.7.1",
    "jspdf": "2.5.1",
    "jspdf-autotable": "3.8.2",
    "marked": "5.1.2",
    "marked-mangle": "1.1.0",
    "ng2-charts": "5.0.4",
    "ngx-cookie-service": "17.1.0",
    "ngx-translate-multi-http-loader": "17.0.0",
    "object-to-formdata": "4.5.1",
    "popper.js": "1.16.1",
    "postcss-preset-env": "9.5.4",
    "primeng": "17.13.0",
    "primeng-sass-theme": "https://github.com/primefaces/primeng-sass-theme",
    "rxjs": "7.8.1",
    "tippy.js": "6.3.7",
    "typescript": "5.4.4",
    "zone.js": "0.14.4"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "17.3.3",
    "@angular-eslint/builder": "17.3.0",
    "@angular-eslint/eslint-plugin": "17.3.0",
    "@angular-eslint/eslint-plugin-template": "17.3.0",
    "@angular-eslint/schematics": "17.3.0",
    "@angular-eslint/template-parser": "17.3.0",
    "@angular/cli": "17.3.3",
    "@angular/compiler-cli": "17.3.3",
    "@angular/language-service": "17.3.3",
    "@types/jasmine": "5.1.4",
    "@types/jasminewd2": "2.0.13",
    "@types/node": "20.12.6",
    "@typescript-eslint/eslint-plugin": "7.6.0",
    "@typescript-eslint/parser": "7.6.0",
    "eslint": "8.56.0",
    "eslint-plugin-import": "2.29.1",
    "eslint-plugin-jsdoc": "48.2.3",
    "eslint-plugin-prefer-arrow": "1.2.3",
    "eslint-plugin-unused-imports": "3.1.0",
    "jasmine-core": "5.1.2",
    "jasmine-spec-reporter": "7.0.0",
    "karma": "6.4.3",
    "karma-chrome-launcher": "3.2.0",
    "karma-coverage": "2.2.1",
    "karma-jasmine": "5.1.0",
    "karma-jasmine-html-reporter": "2.1.0",
    "ngx-translate-testing": "7.0.0",
    "postcss": "8.4.38",
    "protractor": "7.0.0",
    "ts-node": "10.9.2",
    "webdriver-manager": "12.1.9",
    "webpack-bundle-analyzer": "4.10.1"
  },
  "browserslist": "> 5%, last 3 Chrome versions, last 3 Firefox version, last 2 Edge major versions, last 4 Safari major versions, last 4 iOS major versions, Firefox ESR"
}

```

### Reproducer

_No response_

### Angular version

17.3.3

### PrimeNG version

17.13.0

### Build / Runtime

TypeScript

### Language

TypeScript

### Node version (for AoT issues node --version)

20.11.1

### Browser(s)

_No response_

### Steps to reproduce the behavior

_No response_

### Expected behavior

_No response_

### PR DESCRIPTION
**Pull request by BGBRWR**
Currently setting `expandedRowKeys` or `editingRowKeys` with anything will convert them to booleans; Which is the wrong type and breaks working with these inputs.

Fixes #15241 #15264 #15275

### CODE REVIEW

**Comment by BGBRWR** [comment]
@LcsGa

**Comment by LcsGa** [comment]
That might not be the only mistake actually.
There are so many modified lines, it seems like I also made similar mistakes elsewhere (on `trackBy` input functions for exemple).

**Comment by LcsGa** [comment]
I'll take a look at it asap to provide a fix (if nobody did it already).

**Comment by Viktor-Ivliev** [comment]
> That might not be the only mistake actually. There are so many modified lines, it seems like I also made similar mistakes elsewhere (on `trackBy` input functions for exemple).

I think it so https://github.com/primefaces/primeng/issues/15241 `trackBy` really does not work =(

**Comment by BGBRWR** [comment]
I added the remaining fixes to all invalid transform Inputs. It ended up including the same changes in the PR #15248.

**Comment by LcsGa** [comment]
Great! I'm close mine then

**Review by LcsGa — APPROVED** [review]

**Comment by BGBRWR** [comment]
fixes https://github.com/primefaces/primeng/issues/15241

**Review by aaronsalazar — APPROVED** [review]
👍

File: src/app/components/table/table.ts [hunk 1]
```diff
@@ -1944,6 +1944,7 @@ export class Table implements OnInit, AfterViewInit, AfterContentInit, Blockable
     }
 
     isRowSelectable(data: any, index: number) {
+        console.log(data, index);
```
Comment (PaulSchult) [inline src/app/components/table/table.ts:1947]: Is this intended?

Comment (BGBRWR) [reply src/app/components/table/table.ts:1947]: Good catch! Will remove.

**Review by PaulSchult — APPROVED** [review]

**Comment by LcsGa** [comment]
About the last commit, why not simply make a specific transform function instead of completely removing it, as it could increase the DX?

**Comment by PaulSchult** [comment]
> About the last commit, why not simply make a specific transform function instead of completely removing it, as it could increase the DX?

Do you mean something similar to what I suggested in #15264? 

Just curious, how do the input transforms enhance the DX?

**Comment by LcsGa** [comment]
> > About the last commit, why not simply make a specific transform function instead of completely removing it, as it could increase the DX?
> 
> Do you mean something similar to what I suggested in #15264? 
> 
> Just curious, how do the input transforms enhance the DX?

Yes, exactly!
Well, if you know exactly what value you want to pass and it's not meant to be dynamic, you can do things like this with them:
```html
<!-- before -->
<p-inputnumber [minFractionDigit]="2">

<p-button [rounded]="true">

<!-- after -->
<p-inputnumber minFractionDigit="2">

<p-button rounded>
```

As you can see, especially about boolean "flags" it is less code to write and this is one good way for using input transforms. 

Before that transform function, a lot of libraries already provided such things using setters and now it is just easier.

**Comment by BGBRWR** [comment]
I agree but I won't have time to add this improvement. The purpose of this PR is to get this library back to a working state. If this PR is still open a week from now I can try to add custom transforms on InputNumber's minFractionDigit/maxFractionDigit.

**Comment by cetincakiroglu** [comment]
Hi @BGBRWR,

Thanks a lot for the effort and support!

**Pull request merged**